### PR TITLE
Check if document is undefined before using it, Fix phone field width

### PR DESCRIPTION
**Pull request by jordanranz**
_Description of changes:_
- Check for `document` existence

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-ui-components/src/global/theme.ts [hunk 1]
```diff
@@ -11,47 +11,49 @@
...
-
-    --amplify-red: #dd3f5b;
-  }
-`),
-);
-
-let parentElement = document.getElementsByTagName('head')[0];
-const firstChild = parentElement.firstChild;
-parentElement.insertBefore(customStyles, firstChild);
+if (typeof document !== 'undefined') {
```
Comment (ericclemmons) [inline packages/amplify-ui-components/src/global/theme.ts:14]: Should we use `isBrowser` here?
https://github.com/aws-amplify/amplify-js/blob/2f27a6e2d002364af5c44b9f38542ce1e2c8c119/packages/core/src/JS.ts#L159-L171

Comment (ashika01) [reply packages/amplify-ui-components/src/global/theme.ts:14]: Yup we should do that.

Comment (jordanranz) [reply packages/amplify-ui-components/src/global/theme.ts]: done

File: packages/amplify-ui-components/src/components/amplify-button/amplify-button.tsx [hunk 2]
```diff
@@ -32,7 +32,7 @@ export class AmplifyButton {
         const formSection = this.el.closest('amplify-form-section');
         form = formSection && formSection.shadowRoot.querySelector('form');
       }
-      if (form) {
+      if (form && typeof document !== 'undefined') {
```
Comment (jordanranz) [inline packages/amplify-ui-components/src/components/amplify-button/amplify-button.tsx:35]: This needs re-thinking

**Comment by sawyerh** [comment]
Related (but doesn't completely fix): #5293

**Pull request merged**